website/tests/test_leaderboard_scoring.py

```
@@ -0,0 +1,318 @@
+        # Mock new calculation
+        mock_calculate.return_value = (Decimal("75.00"), {"goals": Decimal("80.00")})
+
+        self.user_profile.calculate_leaderboard_score()
+
+        # Verify fields were updated
+        self.user_profile.refresh_from_db()
+        self.assertEqual(self.user_profile.leaderboard_score, Decimal("75.00"))
+        self.assertEqual(self.user_profile.quality_score, Decimal("80.00"))
+        # ensure these fields were refreshed as well
+        self.assertNotEqual(self.user_profile.check_in_count, 5)
+        self.assertGreater(self.user_profile.last_score_update, original_last_score_update)
```

Ruff (0.14.7)

Undefined name `original_last_score_update`

(F821)